### PR TITLE
airbyte-lib: Add path executor

### PR DESCRIPTION
**Pull request by flash1293**
This PR adds a path executor which allows to bypass the venv if the connector is installed locally and available via a cli script.

This behavior can be enabled via setting the `use_venv` param of `get_connector` to false:
```
source = ab.get_connector("source-apify-dataset") # use a venv
source2 = ab.get_connector("source-apify-dataset", use_venv=False) # use `source-apify-dataset` from the path, fails if it doesn't exist
```

### CODE REVIEW

**Comment by stephane-airbyte** [comment]
@flash1293 I think you ran into a bad merge. You'll have to close this PR, rebase your branch on master and recreate a PR (if you just rebase your branch, everyone currently on this PR will be ping everytime something happens here)

**Comment by aaronsteers** [comment]
@flash1293 - When I tried to use Graphite on my own PRs (based on the same base as this one), it force pushed all 3 branches, and may have been the cause of this one getting into a bad state.

I updated the base from master and this one from the base branch, and it looks okay now.

File: airbyte-lib/airbyte_lib/factories.py [hunk 1]
```diff
@@ -16,9 +16,16 @@ def get_in_memory_cache():
...
 
-def get_connector(name: str, version: str = "latest", config: Optional[Dict[str, Any]] = None):
+def get_connector(name: str, version: str = "latest", config: Optional[Dict[str, Any]] = None, use_venv: bool = True):
+    """
+    Get a connector by name and version.
+    :param name: connector name
+    :param version: connector version - if not provided, the most recent version will be used
+    :param config: connector config - if not provided, you need to set it later via the set_config method
+    :param use_venv: whether to use a virtual environment to run the connector. If False, the connector is expected to be available on the path (e.g. installed via pip). If True, the connector will be installed in a virtual environment.
+    """
```
Comment (aaronsteers) [inline airbyte-lib/airbyte_lib/factories.py:26]: I read "use_venv" the first time to mean that "false" would still be managed, but would be installed by AirbyteLib without venv isolation. Wdyt of a name like "already_installed" or "needs_install" or similar to distinguish AirbyteLib-managed installations vs user-managed?

**Review by aaronsteers — APPROVED** [review]
One question inline, but non-blocking. 

Approved for merge. 👍

File: airbyte-lib/airbyte_lib/factories.py (hunk 1 above)
Comment (flash1293) [reply airbyte-lib/airbyte_lib/factories.py:26]: what about "auto_install"?

Comment (flash1293) [reply airbyte-lib/airbyte_lib/factories.py:26]: I merged already, I will implement this on the base PR

**Pull request merged**